xdsl/transforms/individual_rewrite.py

```diff
@@ -0,0 +1,37 @@
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.tools.command_line_tool import get_all_dialects
+from xdsl.traits import HasCanonicalisationPatternsTrait
+
+REWRITE_BY_NAMES: dict[str, dict[str, RewritePattern]] = {
```

Comment:
hmm, this feels like the optimization machinery within xDSL should be a bit more introspectable by construction.
I understand this is the minimum effort that can be done ATM without overhauling that part of the framework.

Maybe it's worth having it as a discussion some time soon.

Reply:
Agree. Would you like to join this Thursday's weekly interactive meeting to talk about it? :) It's at 14:00 UK time.